appstart/sandbox/container_sandbox.py

```diff
@@ -464,8 +467,19 @@ def exit_loop_with_error(error):
             attempt += 1
             time.sleep(1)
 
+        # Tell the user where to connect, depending on whether the
+        # devappserver is running.
+        if self.run_devappserver:
+            port = self.proxy_port
+        else:
+            port = self.port
         get_logger().info('Your application is live. '
-                          'Access it at: {0}:{1}'.format(host, str(self.port)))
+                          'Access it at: {0}:{1}'.format(host, port))
+        if self.run_devappserver:
+            get_logger().info('(port {0} goes through the dev_appserver '
```

Comment:
It makes for a more user-friendly message flow without duplicating the log message.